props/saber_fett263_buttons.h

```diff
@@ -293,166 +664,3131 @@ class SaberFett263Buttons : public virtual PropBase {
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<Int<7000>,Int<-6000>>,RotateColorsX<Variation,RgbArg<1,Rgb<255,0,0>>>,ShowColorStyle>> hilt_color_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<Int<7000>,Int<-6000>>,Black,RotateColorsX<Variation,ShowColorStyle>>> pre_color_;
+  ShowColorSingleBladeTemplate<LengthEdit<RotateColorsX<Variation,ShowColorStyle>,Pulsing<White,Rgb<100,100,100>,800>>> show_length_;
+  ShowColorSingleBladeTemplate<Mix<Bump<IntEdit,Int<10000>>,Black,ShowColorStyle>> show_lockup_position_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<IntEdit,Int<6000>>,Black,ShowColorStyle>> show_drag_size_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<IntEdit,Int<-6000>>,Black,ShowColorStyle>> show_emitter_size_;
+#if NUM_BLADES > 1
+  ShowColorSingleBladeTemplate<Pulsing<RotateColorsX<Variation,ShowColorStyle>,Black,800>> show_preview_;
 #endif
-#ifdef FETT263_BATTLE_MODE_START_ON
-          battle_mode_ = true;
+
+  Color16 GetColorArg(int blade, int arg) {
+    char argspace[32];
+    if (style_parser.GetArgument(current_preset_.GetStyle(blade), arg + 2, argspace)) {
+      char* tmp;
+      int r = strtol(argspace, &tmp, 0);
+      int g = strtol(tmp+1, &tmp, 0);
+      int b = strtol(tmp+1, NULL, 0);
+      return Color16(r,g,b);
+    }
+    return Color16();
+  }
+
+  // Show Full Blade Preview for Color Editing
+  void ShowFull() {
+    show_color_.Start(blade_num_);
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, effect_num_)); 
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl_ = saved_color_.toHSL();
+    hsl_angle_ = fusor.angle2();
+  }
+
+  // Show Partial Blade Preview (Clash, Lockup, Drag, Melt, Preon, Emitter) for Color Editing
+  void ShowPart() {
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, effect_num_)); 
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl_ = saved_color_.toHSL();
+    hsl_angle_ = fusor.angle2();
+  }
+
+#if NUM_BLADES > 1
+  void BladePreview(int blade) {
+    if (blade_preview_ > 0) {
+      show_preview_.Stop(blade_preview_);
+    }
+    blade_preview_ = blade;
+    show_preview_.Start(blade);
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, 1)); 
+  }
 #endif
-        }
-        return true;
 
-#ifdef BLADE_DETECT_PIN
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_ON):
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_OFF):
-        // Might need to do something cleaner, but let's try this for now.
-        blade_detected_ = true;
-        FindBladeAgain();
-        SaberBase::DoBladeDetect(true);
-        return true;
+  void LengthPreview(int blade) {
+    show_length_.Start(blade);
+    length_edit_length = blade_length_ - 1;
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, 1)); 
+  }
+ 
+  // Copy Color Arguments from one blade to another
+  void SetColorCopy() {
```

Comment:
I'm not sure why this is called _Set_ColorCopy.
What is it setting?
If it's setting something, why doesn't the comment say so?

Reply:
renamed to CopyColors()